monitor/plugins/spark_progress.py

```diff
@@ -30,58 +30,103 @@
 class SparkProgress(Plugin):
 
     def __init__(self, app_id, info_plugin, collect_period, retries=60):
-        Plugin.__init__(self, app_id, info_plugin, collect_period, retries=retries)
+        Plugin.__init__(self, app_id, info_plugin,
+                        collect_period, retries=retries)
+
+        self.monasca = MonascaMonitor()
+
         self.submission_url = info_plugin['spark_submisson_url']
         self.expected_time = info_plugin['expected_time']
-        self.monasca = MonascaMonitor()
-        self.dimensions = {'application_id': self.app_id, 'service': 'spark-sahara'}
+
+        self.remaining_time = int(self.expected_time)
+        self.job_time = int(self.expected_time)
+
+        self.number_of_jobs = int(info_plugin['number_of_jobs'])
+        self.current_job_id = 0
+
+        self.dimensions = {'application_id': self.app_id,
+                           'service': 'spark-sahara'}
+
+
+    def _publish_measurement(self, job_request):
+
+        application_progress_error = {}
+
+        # Init
+        jobs = job_request.json()
+        jobs.reverse()
+
+        if not len(jobs) == 0:
+            current_job = jobs[self.current_job_id]
+            
+            if current_job['status'] == 'SUCCEEDED':
+                elapsed_time = float(self._get_elapsed_time(
+                               current_job['submissionTime']))            
+
+                self.remaining_time = self.remaining_time - elapsed_time
+
+                self.current_job_id = len(jobs) - 1
+
+                # Job Time
+                self.job_time = (self.remaining_time
```

Comment:
I think you should rename this to "job_expected_time". It is easier to understand.

Reply:
Agree.

Reply:
@robertonscjr please fix apply this suggestion